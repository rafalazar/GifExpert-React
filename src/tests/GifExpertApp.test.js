import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../../src/GifExpertApp';

describe('Pruebas sobre <GifExpertApp />', () => {
    
    
    test('Debe mostrar el componente correctamente', () => {
        
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar una lista de categorÃ­as', () => {
       
        const categories = ['The Last of Us', 'God of war'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
        
    });
    
});
