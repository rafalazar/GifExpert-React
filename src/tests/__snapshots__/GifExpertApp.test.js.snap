// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifExpertApp /> Debe de mostrar una lista de categor√≠as 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="The Last of Us"
      key="The Last of Us"
    />
    <GifGrid
      category="God of war"
      key="God of war"
    />
  </ol>
</div>
`;

exports[`Pruebas sobre <GifExpertApp /> Debe mostrar el componente correctamente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
